spec/requests/api/v1/journeys_controller_create_spec.rb

```diff
@@ -0,0 +1,147 @@
+              "data": {
+                "id": to_location_id,
+                "type": 'locations',
+              },
+            },
+          },
+        },
+      }
+    }
+
+    before do
+      post "/api/v1/moves/#{move_id}/journeys", params: journey_params, headers: headers, as: :json
+    end
+
+    context 'when successful' do
+      let(:schema) { load_yaml_schema('post_journeys_responses.yaml') }
+      let(:data) {
+        {
+          "id": Journey.first&.id,
+          "type": 'journeys',
+          "attributes": {
+            "billable": false,
+            "state": 'in_progress',
+            "timestamp": '2020-05-04T09:00:00+01:00',
+            "vehicle": { "id": '12345678ABC', "registration": 'AB12 CDE' },
+          },
+          "relationships": {
+            "from_location": {
+              "data": {
+                "id": from_location_id,
+                "type": 'locations',
+              },
+            },
+            "to_location": {
+              "data": {
+                "id": to_location_id,
+                "type": 'locations',
+              },
+            },
+          },
+        }
+      }
+
+      it_behaves_like 'an endpoint that responds with success 201'
+
+      it 'returns the correct data' do
+        expect(response_json).to include_json(data: data)
+      end
+    end
+
+    context 'when unsuccessful' do
+      let(:schema) { load_yaml_schema('error_responses.yaml') }
+
+      context 'with a bad request' do
+        let(:journey_params) { nil }
+
+        it_behaves_like 'an endpoint that responds with error 400'
+      end
+
+      context 'with an invalid timestamp' do
+        let(:timestamp) { 'foo-bar' }
+
+        it_behaves_like 'an endpoint that responds with error 400' do
+          let(:errors_400) {
+            [{ 'title' => 'Invalid timestamp',
+               'detail' => 'Validation failed: Timestamp must be formatted as a valid ISO-8601 date-time' }]
+          }
+        end
+      end
+
+      context 'with an invalid billable' do
+        let(:billable) { 'foo-bar' }
+
+        it_behaves_like 'an endpoint that responds with error 400' do
+          let(:errors_400) {
+            [{ 'title' => 'Invalid billable',
+               'detail' => 'Validation failed: Billable is not included in the list' }]
+          }
+        end
+      end
+
+      context 'when not authorized' do
+        let(:access_token) { 'foo-bar' }
+        let(:detail_401) { 'Token expired or invalid' }
+
+        it_behaves_like 'an endpoint that responds with error 401'
+      end
+
+      context 'with a missing move_id' do
+        let(:move_id) { 'foo-bar' }
+        let(:detail_404) { "Couldn't find Move with 'id'=foo-bar" }
+
+        it_behaves_like 'an endpoint that responds with error 404'
+      end
+
+      context 'with a reference to a missing relationship' do
+        let(:to_location_id) { 'foo-bar' }
+        let(:detail_404) { "Couldn't find Location with 'id'=foo-bar" }
```

Comment:
this is a POST to the journey,
I am not sure if `404 not found` is the best status to return.
I've read that it should be used when we are trying to get a resource, the 422 (Unprocessable Entity) might be more appropriate, but double check with Jonny about this.

Reply:
now returns 422 for missing referenced location 👍